lib/connect_client.go

```diff
@@ -325,6 +314,17 @@ func (p connectClient) sync(ctx context.Context, logger DatabaseLogger, tableNam
 			}
 		}
 
+		if res.Cursor != nil {
+			tc = res.Cursor
+		}
+
+		// Because of the ordering of events in a vstream
+		// we receive the vgtid event first and then the rows.
```

Comment:
This comment is incorrect. there's a bit of nuance to it, but generally the `vgtid` event comes _after_ row data. Not asking you to do anything different in this PR, just making note of the issue.